src/test/java/com/google/roomies/servlets/ListingsServletTest.java

```diff
@@ -76,6 +88,53 @@ public void setUp() throws Exception {
 
     DatabaseFactory.setDatabaseForTest(db);
     setRequestParameters();
+    defineDatabaseMockBehavior();
+
+    stringWriter = new StringWriter();
+    when(response.getWriter()).thenReturn(new PrintWriter(stringWriter));
+  }
+ 
+  @Test
+  public void testGet_returnsSingleListing() throws Exception {
+    Map<String, Object> listingData = mapOfListingDataForGetTests(request);
+    when(queryDocumentMock.getData()).thenReturn(listingData);
+ 
+    listingsServlet.doGet(request, response);
+    String expectedWriterOutput =  "[{\"documentId\":{\"value\":\"documentID\"}," + 
+      "\"timestamp\":{\"value\":{\"seconds\":1474156800,\"nanos\":0}},\"title\":\"Test title\"," +
+      "\"description\":\"Test description\",\"startDate\":\"Jul 10, 2020, 12:00:00 AM\"," + 
+      "\"endDate\":\"Jul 10, 2020, 12:00:00 AM\",\"leaseType\":\"YEAR_LONG\",\"numRooms\":2," +
+      "\"numBathrooms\":2,\"numShared\":2,\"numSingles\":2,\"sharedPrice\":\"USD 100\","+
+      "\"singlePrice\":\"USD 0\",\"listingPrice\":\"USD 100\"}]";
+
+    verify(db, Mockito.times(1)).getAllDocumentsInCollection(LISTING_COLLECTION_NAME);
+    verify(queryDocumentMock, Mockito.times(1)).getData();
+    assertEquals(stringWriter.getBuffer().toString().trim(), expectedWriterOutput);
+  }
+
+  @Test
+  public void testGet_returnsNoListingsWithNoneInFirestore() throws Exception {
+    List<QueryDocumentSnapshot> queryDocumentList = new ArrayList();
+    when(queryDocumentsMock.spliterator()).thenReturn(queryDocumentList.spliterator());
+
+    listingsServlet.doGet(request, response);
+
+    verify(db, Mockito.times(1)).getAllDocumentsInCollection(LISTING_COLLECTION_NAME);
+    verify(queryDocumentMock, Mockito.times(0)).getData();
```

Comment:
Again, either unneeded or distinct behavior (can be some debate on this)

Reply:
Removed all verify tests from doGet() tests. I think comparing the response output to the expected output is sufficient.